### PR TITLE
Round15: grad-clip-norm=0.5 — halve gradient clip (SOTA=1.0, never varied)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The gradient clip norm (`--grad-clip-norm`) has been fixed at **1.0 across all ~60 tay experiments** and has never been varied. This is a virgin lever.

**Why it matters for Lion:** Lion uses a sign-based gradient update — the raw gradient is normalized to its sign before the weight update. But gradient clipping happens _before_ the sign operation (in `trainer_runtime.py`'s call to `torch.nn.utils.clip_grad_norm_`). Clipping at 0.5 means we only clip gradients whose L2 norm exceeds 0.5, reducing the step variance contributed by large-norm gradient events.

**Theoretical motivation:**
- Tighter clipping reduces the influence of outlier gradient steps (e.g. on turbulent boundary layers where pressure gradients are large)
- Combined with Lion's sign update, tighter clipping creates a more stable directional signal across all GPUs in DDP
- Late-epoch benefit: when the loss landscape is flat and small gradient signals dominate, noisier high-norm gradients (that get clipped at 1.0 but were still large pre-clip) will be eliminated

**Single delta vs SOTA:** `--grad-clip-norm 0.5` (SOTA: `--grad-clip-norm 1.0`, never varied)

## Instructions

Run the following command **exactly as written** (single delta — only `--grad-clip-norm 0.5` changes from SOTA):

```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent haku --wandb-name "haku/round15-grad-clip-0p5" \
  --wandb-group "tay-round15-grad-clip" \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --ema-decay 0.999 --lr-warmup-epochs 1 \
  --grad-clip-norm 0.5
```

Once launched, post your W&B run IDs (rank0 + all 8 ranks) as a comment on this PR.

Report `val_primary/abupt_axis_mean_rel_l2_pct` at each epoch as it completes.

When training finishes, report all `test_primary/*` metrics.

**Watch for:** If tighter clipping causes training instability (sudden val spikes, divergence), report immediately — it would mean the 1.0 clip is load-bearing.

## Baseline

Current SOTA: **PR #222 fern** (`ut1qmc3i`, group `tay-round12-lr-warmup-1ep`)

| Epoch | val_abupt |
|-------|-----------|
| ep7   | 9.8759%   |
| ep8   | 9.4516%   |
| **ep9** | **9.2910%** |

| Metric | PR #222 (current SOTA) | AB-UPT target |
|--------|------------------------|---------------|
| `test_abupt_axis_mean_rel_l2_pct` | **10.420%** | — |
| `test_surface_pressure_rel_l2_pct` | **5.550%** | 3.82% |
| `test_wall_shear_rel_l2_pct` | **10.185%** | 7.29% |
| `test_volume_pressure_rel_l2_pct` | **12.737%** | 6.08% |
| `test_wall_shear_x_rel_l2_pct` | **8.629%** | 5.35% |
| `test_wall_shear_y_rel_l2_pct` | **12.329%** | 3.65% |
| `test_wall_shear_z_rel_l2_pct` | **12.854%** | 3.63% |

**Reproduce SOTA (your delta is only --grad-clip-norm 0.5):**
```bash
torchrun --standalone --nproc_per_node=8 train.py \
  --agent <STUDENT> --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --ema-decay 0.999 --lr-warmup-epochs 1
```

**Merge bar:** `val_primary/abupt_axis_mean_rel_l2_pct` < **9.291%** (PR #222 best val ep9).

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Status check — this PR has been assigned for 24+ hours with no activity comment. Please confirm:

1. Have you launched the training run? Post your W&B run ID (rank0) as a reply to this comment.
2. If you hit any issues (OOM, import error, flag not recognized), describe them immediately.

Your experiment: `--grad-clip-norm 0.5` (virgin lever — never tested in 60+ runs). The run command is in the PR body. Launch it now if you haven't already.

**Comment by morganmcg1** [comment]
ADVISOR: Second escalation. You were assigned this experiment over 24 hours ago and no W&B activity has been detected. SOTA is 9.065% and we need these experiments running immediately. If you are having environment or pod issues, reply with the error and I will help unblock you. Otherwise, start training NOW and post your W&B run ID in this PR within the next 2 hours. GPU time is precious — if there is no response within 2 hours, this PR will be reassigned.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #267 because ADVISOR: Closing — this PR was assigned to a student without a DrivAerML pod (this cluster only has alphonse/askeladd/edward/fern/frieren/nezuko/tanjiro/thorfinn). The hypothesis will be re-issued to an actual DrivAerML student in a fresh PR..
